javalin/src/main/java/io/javalin/plugin/PluginExceptions.kt

```diff
@@ -5,3 +5,8 @@ abstract class PluginException(pluginClass: Class<out Plugin<*>>, override val m
 
 data class PluginAlreadyRegisteredException(val plugin: Plugin<*>) :
     PluginException(plugin::class.java, "${plugin.name()} is already registered")
+
+data class PluginKeyAlreadyRegisteredException(val plugin: Plugin<*>) :
+    PluginException(plugin::class.java, "${plugin.name()} is already registered with the same key")
+
+class PluginNotRegisteredException : RuntimeException("Requested plugin was not registered at startup")
```

Comment:
`PluginNotRegisteredException` could extend `IllegalStateException`, which is a bit more specific. It should also include information about which plugin was not registered.